File: src/main/java/io/srmppn/HappyDorm/billing/api/BillCancellation.java
package io.srmppn.HappyDorm.billing.api;

import org.axonframework.modelling.command.TargetAggregateIdentifier;

import java.util.UUID;

public class BillCancellation {
    public static class CancelBillCommand {
        @TargetAggregateIdentifier
        public UUID billId;

        public CancelBillCommand(UUID billId) {
            this.billId = billId;
        }

        public UUID getBillId() {
            return billId;
        }

        public void setBillId(UUID billId) {
            this.billId = billId;
        }
    }

    public static class BillCancelledEvent {
        @TargetAggregateIdentifier
        public UUID billId;

        public BillCancelledEvent(UUID billId) {
            this.billId = billId;
        }

        public UUID getBillId() {
            return billId;
        }

        public void setBillId(UUID billId) {
            this.billId = billId;
        }
    }
}

File: src/main/java/io/srmppn/happydorm/billing/api/bill-cancellation.kt
package io.srmppn.happydorm.billing.api

import org.axonframework.modelling.command.TargetAggregateIdentifier
import org.axonframework.serialization.Revision
import java.util.*

data class CancelBillCommand(
  @TargetAggregateIdentifier
  val billId: UUID
)

@Revision("1.0")
data class BillCancelledEvent(
  @TargetAggregateIdentifier
  val billId: UUID
)